lib/AST/ASTContext.cpp

```diff
@@ -8059,6 +8080,11 @@ QualType ASTContext::mergeFunctionTypes(QualType lhs, QualType rhs,
       }
     }
 
+    // Compatible functions must have the same number of type variables
+    // FIXME : Make sure to pass correct QualType.
```

Comment:
The FIXME is wrong. I must have mistaken it for being the code that is doing compatibility check. The FIXME should be removed.